timesync/clock_test.go

```diff
@@ -61,43 +55,10 @@ func TestTicker_StartClock_BeforeEpoch(t *testing.T) {
 	ts.Close()
 }
 
-func TestTicker_StartClock_LayerID(t *testing.T) {
-	tick := 1 * time.Second
-	layout := "2006-01-02T15:04:05.000Z"
-	str := "2018-11-12T11:45:20.371Z"
-	start, _ := time.Parse(layout, str)
-
-	ts := NewTicker(MockTimer{}, tick, start)
-	ts.updateLayerID()
-	assert.Equal(t, types.LayerID(8), ts.nextLayerToTick)
-	ts.Close()
-}
-
-func TestTicker_StartClock_2(t *testing.T) {
-	destTime := 2 * time.Second
-	tmr := &RealClock{}
-	then := tmr.Now()
-	ticker := NewTicker(tmr, 5*time.Second, then.Add(destTime))
-	ticker.StartNotifying()
-	sub := ticker.Subscribe()
-	<-sub
-	assert.True(t, tmr.Now().Sub(then).Seconds() <= float64(2.1))
-}
-
-func TestTicker_Tick(t *testing.T) {
+func TestClock_TickFutureGenesis(t *testing.T) {
 	tmr := &RealClock{}
-	ticker := NewTicker(tmr, 5*time.Second, tmr.Now())
-	ticker.started = true
-	ticker.notifyOnTick()
-	l := ticker.nextLayerToTick
-	ticker.notifyOnTick()
-	assert.Equal(t, ticker.nextLayerToTick, l+1)
-}
-
-func TestTicker_TickFutureGenesis(t *testing.T) {
-	tmr := &RealClock{}
-	ticker := NewTicker(tmr, 1*time.Second, tmr.Now().Add(2*time.Second))
-	assert.Equal(t, types.LayerID(1), ticker.nextLayerToTick) // check assumption that nextLayerToTick >= 1
+	ticker := NewClock(tmr, 1*time.Second, tmr.Now().Add(2*time.Second))
+	assert.Equal(t, types.LayerID(1), ticker.lastTickedLayer+1) // check assumption that nextLayerToTick >= 1
```

Comment:
this assertion is weird... `assert that 1 == something + 1`. Please also update the comment.

Reply:
fixed